### ISSUE TITLE
Plugins Class `PluginsDirectory` ArgumentNullException

### ISSUE THREAD

**Issue by cschuchardt88** [neo-project/neo#2943]
**Describe the bug**
```
PS C:\Users\Chris\Projects\cschuchardt88\neo-express\src\neoxp\bin\Debug\net7.0\win-x64\publish> .\neoxp.exe run
System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at Neo.Plugins.Plugin..cctor()
```

https://github.com/neo-project/neo/blob/3fc889509ff27adffb17d37c9974ff1815deb255/src/Neo/Plugins/Plugin.cs#L35

**To Reproduce**
See https://github.com/neo-project/neo-express/issues/268

**Expected behavior**
We are trying to use self-contained, so i believe `Assembly.GetEntryAssembly().Location` is null.
![image](https://github.com/neo-project/neo/assets/8141309/f227b725-a2c7-4c84-93d5-d90141557748)

>The [GetEntryAssembly](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-7.0) method can return null when a managed assembly has been loaded from an unmanaged application. For example, if an unmanaged application creates an instance of a COM component written in C#, a call to the [GetEntryAssembly](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-7.0) method from the C# component returns null, because the entry point for the process was unmanaged code rather than a managed assembly.

Use `Environment.CurrentDirectory` instead

**Platform:**
 - Version 3.6.0

**Comment by Liaojinghui** [comment]
Why this is a bug? It works fine if there is a Plugins folder

**Comment by cschuchardt88** [comment]
Its has to do with the initialization of a static Plugin Class. The above error is from `System.TypeInitializationException: The type initializer for 'Neo.Plugins.Plugin' threw an exception.` I do believe when creating a self-contained single file publish for `neo-express` its calling from unmanaged code, witch will make the  assembly entry be null. This is why we are getting that error.

**Comment by cschuchardt88** [comment]
https://github.com/neo-project/neo/pull/2945 is a better solution

**Comment by shargon** [comment]
Solved by #2946
